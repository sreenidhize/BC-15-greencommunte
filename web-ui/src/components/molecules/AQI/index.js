import { Grid } from '@material-ui/core';
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Counter from '../../atoms/Counter';
import Label from '../../atoms/Label';

const useStyles = makeStyles((theme) => ({
    root: {
        height: '100vh',
        textAlign: 'center',
        backgroundColor: "#e7fce0",
    },
    imageItem: {
        // marginTop: theme.spacing(-10),
        marginBottom: theme.spacing(7),
    },
    label : {
        // width: "490px",
    },
    image: {
        height: "auto",
    },
}));

const AQI = (props) => {
    const classes = useStyles();
    const {src, alt, totalJobs, city, size, text, dataTestId } = props;
    return (
        <Grid container className={classes.root} direction='column' justify='center'
        alignItems='center' spacing={2} data-testid={dataTestId}>
            <Grid item className={classes.imageItem}>
                <img src={src} alt={alt} className={classes.image} />
            </Grid>
                <Grid item>
                    <Counter totalJobs={totalJobs} city={city}></Counter>
                </Grid>
            <Grid item className={classes.label}>
                <Label size={size} text={text} ></Label>
            </Grid>
        </Grid>
    );
};

export default AQI;