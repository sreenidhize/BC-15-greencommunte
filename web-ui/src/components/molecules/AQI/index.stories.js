import React from "react";
import AQI from ".";

export default {
  title: "molecules/AQI",
  component: AQI,
};

const Template = (args) => <AQI {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  src: 'assets/one.png',
  alt: "green commute",
  size: "large",
  text: "Enter the location to know Time Air Quality Index (AQI)"
};

export const Secondary = Template.bind({});
Secondary.args = {
  src: 'assets/one.png',
  alt: "green commute",
  totalJobs: 894,
  city: "Hyderabad",
  size: "small",
  text: "Real- Time Air Quality Index(AQI) in this location"
};