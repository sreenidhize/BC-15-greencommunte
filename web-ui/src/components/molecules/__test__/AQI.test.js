import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import AQI from "../AQI";

const src ="https://cdn.zeplin.io/5fa28fcbff79c58958e4e1a6/assets/2fbdf0a6-8c92-401e-9811-0d95634962fa.svg";
const alt = "green commute";
const totalJobs = 894;
const city = "Hyderabad";
const size = "small";
const text = "Real- Time Air Quality Index(AQI) in this location";
const testId = "AQI";

describe("AQI test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <AQI src={src} alt={alt} totalJobs={totalJobs} city={city} size={size} text={text} dataTestId={testId} />
    );
    expect(container).toMatchSnapshot();
  });

  test("test AQI render", () => {
    const { getByTestId } = render(<AQI src={src} alt={alt} totalJobs={totalJobs} city={city} size={size} text={text} dataTestId={testId} />);
    expect(getByTestId("AQI")).toBeInTheDocument();
  });
});