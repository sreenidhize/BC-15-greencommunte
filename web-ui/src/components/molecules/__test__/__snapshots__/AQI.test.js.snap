// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AQI test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    data-testid="AQI"
  >
    <div
      class="MuiGrid-root makeStyles-imageItem-2 MuiGrid-item"
    >
      <img
        alt="green commute"
        class="makeStyles-image-4"
        src="https://cdn.zeplin.io/5fa28fcbff79c58958e4e1a6/assets/2fbdf0a6-8c92-401e-9811-0d95634962fa.svg"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        style="justify-content: center;"
      >
        <div>
          <h3
            class="MuiTypography-root makeStyles-jobs-5 MuiTypography-h3"
          >
            894
          </h3>
        </div>
        <div>
          <h5
            class="MuiTypography-root makeStyles-city-6 MuiTypography-h5"
          >
            Hyderabad
          </h5>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-label-3 MuiGrid-item"
    >
      <div>
        <p
          class="MuiTypography-root makeStyles-small-7 MuiTypography-body2"
          data-testid="small-text"
        >
          Real- Time Air Quality Index(AQI) in this location
        </p>
      </div>
    </div>
  </div>
</div>
`;
